src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlExceptionTest.cs

```diff
@@ -33,6 +33,29 @@ public void SerializationTest()
             Assert.Equal(e.StackTrace, sqlEx.StackTrace);
         }
 
+#if !NET5_0
```

Comment:
This should probably either be handled in the csproj with a condition or use `[SkipOnTargetFramework]`, like the other tests

Reply:
`NET5_0` is a defined constant of the NetCore directives. The preferred is  to not run on `NET 5` only and I believe there is no specific `[SkipOnTargetFramework]` to emphasis on version, please correct me if I am wrong. I have a plan to try and check if it could be added to XUnitExtensions in some free time intervals in future. Adding a constant in csproj is another option, but what would be the difference between adding a new constant to csproj or use predefined preprocessor value? Is there any benefit doing so?

All being said,  I don't see any issue with disabling the test on NetCore as we have other kind of serialization options there.

Reply:
We also want to ensure we don't end up running this on .NET 6.0 in future, so having something for ".NET 5 and above" is what we should include.

I see a point in using [SkipOnTargetFramework] to categorize .NET 5 and further to a totally new [Moniker](https://github.com/dotnet/SqlClient/blob/6a9e4f104ce77ba631e2ac8ea2e366a4d41ba235/src/Microsoft.Data.SqlClient/tests/tools/Microsoft.DotNet.XUnitExtensions/TargetFrameworkMonikers.cs) (e.g. Net=0x16)?

Reply:
> I see a point in using [SkipOnTargetFramework] to categorize .NET 5 and further to a totally new Moniker (e.g. Net=0x16)? It does get complex later, since later we will also have new features/tests for new versions, and running / skipping them with moniker / target framework would be something we'll have to toss often.

I totally agree. This is on my radar to be added. Let me check if it could be a quick amendment to our XUnitExtensions . It will be needed as `.NET` is progressing to a different direction than `NetCoreapps` and `NetFramework`.